src/Browser.php

```diff
@@ -725,6 +728,72 @@ public function withResponseBuffer($maximumSize)
         ));
     }
 
+    /**
+     * Add a request header for all following requests.
+     *
+     * ```php
+     * $browser = $browser->withHeader('User-Agent', 'ACME');
+     *
+     * $browser->get($url)->then(function (Psr\Http\Message\ResponseInterface $response) {
+     * //will contain the header User-Agent with the value ACME
+     * var_dump($response);
+     * }, function (Exception $e) {
+     * echo 'Error: ' . $e->getMessage() . PHP_EOL;
+     * });
+     * ```
```

Comment:
Nit, to match the example in `README.md`:
```suggestion
     * ```php
     * $browser = $browser->withHeader('User-Agent', 'ACME');
     *
     * $browser->get($url)->then(function (Psr\Http\Message\ResponseInterface $response) {
     *     //will contain the header User-Agent with the value ACME
     *     var_dump($response);
     * }, function (Exception $e) {
     *     echo 'Error: ' . $e->getMessage() . PHP_EOL;
     * });
     * ```
```